controllers/hardwareclassification_controller.go

```diff
@@ -33,24 +41,68 @@ type HardwareClassificationReconciler struct {
 	Scheme *runtime.Scheme
 }
 
+// Reconcile reconcile function
 // +kubebuilder:rbac:groups=metal3.io,resources=hardwareclassifications,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=metal3.io,resources=hardwareclassifications/status,verbs=get;update;patch
+func (hcReconciler *HardwareClassificationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+
+	// Initialize the logger with namespace
+	hcReconciler.Log = hcReconciler.Log.WithName(HWControllerName).WithValues("metal3-harwdwareclassification", req.NamespacedName)
+
+	// Get HardwareClassificationController to get values for Namespace and ExpectedHardwareConfiguration
+	hardwareClassification := &hwcc.HardwareClassification{}
+
+	if err := hcReconciler.Client.Get(ctx, req.NamespacedName, hardwareClassification); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	// Get ExpectedHardwareConfiguraton from hardwareClassification
+	extractedProfile := hardwareClassification.Spec.ExpectedHardwareConfiguration
+	hcReconciler.Log.Info("Extracted hardware configurations successfully", "Profile", extractedProfile)
+
+	// Get the new hardware classification manager
+	hcManager := utils.NewHardwareClassificationManager(hcReconciler.Client, hcReconciler.Log)
+
+	ErrValidation := hcManager.ValidateExtractedHardwareProfile(extractedProfile)
+	if ErrValidation != nil {
+		hcReconciler.Log.Error(ErrValidation, ErrValidation.Error())
+		hcReconciler.handleErrorConditions(req, hardwareClassification, hwcc.ProfileMisConfigured, ErrValidation.Error(), hwcc.ProfileMatchStatusEmpty)
+		return ctrl.Result{}, nil
+	}
 
-// Add RBAC rules to access baremetalhost resources
-// +kubebuilder:rbac:groups=metal3.io,resources=baremetalhosts,verbs=get;list;update
-// +kubebuilder:rbac:groups=metal3.io,resources=baremetalhosts/status,verbs=get
```

Comment:
If you do not have this anywhere, you would lose the part of the RBAC role that gives you rights on the BMH

Reply:
@Ashughorla  RBAC's are important for accessing the other resources. Please don't remove above changes.

Reply:
@digambar15  Okay, will add RBAC rules and update the PR.